react/features/base/conference/middleware.web.ts

```diff
@@ -32,8 +58,15 @@ MiddlewareRegistry.register(store => next => action => {
             dispatch(setSkipPrejoinOnReload(true));
         }
 
+        releaseScreenLock();
+
         break;
     }
+    case CONFERENCE_LEFT:
+    case KICKED_OUT:
+        releaseScreenLock();
```

Comment:
It seems to me that kick is not touching the conference. I'd prefer to leave it as it is.

Shouldn't the order for the breakout room be leave and then join? wouldn't we mess up the conference object in redux if this is not the order?

Reply:
I'd like us to think about consistency. Web is more forgiving because the page is reloaded with every meeting, not so on mobile. The state listener patter is pretty wide spread across the code base at this point.

Reply:
- I don't think there's an issue with consistency because middlewares on those actions are widely used.
 - The feature is specific to web
 - The breakout rooms events will arrive in the correct order see - https://github.com/jitsi/jitsi-meet/blob/6a21a7e74ba35374c3834a479454ca315973a0ad/react/features/breakout-rooms/actions.ts#L229
 - In order to implement it the way you are asking I need to start clearing the conference state on kick. Let's not over-engineer this otherwise simple PR!